magenta-lib/src/main/scala/magenta/graph/SimpleGraph.scala

```diff
@@ -0,0 +1,101 @@
+package magenta.graph
+
+sealed trait Node[+T] {
+  def ~>[R >: T](to: Node[R]): Edge[R] = Edge(this, to)
+  def maybePriority: Option[Int] = None
+  def maybeValue: Option[T] = None
+}
+case object StartNode extends Node[Nothing]
+case class MidNode[T](value: T, priority: Int = 1) extends Node[T] {
+  override def maybePriority: Option[Int] = Some(priority)
+  def incPriority(n: Int): MidNode[T] = this.copy(priority = priority + n)
+  override def maybeValue: Option[T] = Some(value)
+}
+case object EndNode extends Node[Nothing]
+
+case class Edge[+T](from: Node[T], to: Node[T])
+
+object SimpleGraph {
+  def apply[T](edges: Edge[T]*): SimpleGraph[T] = SimpleGraph(edges.toSet)
+
+  def empty[T] = SimpleGraph[T](StartNode ~> EndNode)
+}
+
+case class SimpleGraph[T](edges: Set[Edge[T]]) {
```

Comment:
My prediction: one day this class will grow into a 1000-line monster and you will rue the day you called it `SimpleGraph`